### DISCUSSION TITLE
[1.18+] Addon Requests

### DISCUSSION THREAD

**Discussion by NovaMachina** [NovaMachina-Mods/ExNihiloSequentia#316] [Ideas]
Over the past several months, our team has seen many users request addons for various mods. Thanks to the data pack and tag systems put in place by Mojang and the Minecraft Forge team, we do not have to create and addon to be compatible with every mod that exists. We will however discuss creating addons for mods that meet at least one of the following criteria:

- Have a custom ore type that is not included in vanilla Ex Nihilo: Sequentia (i.e., Thermal Foundation or Tinker's Construct)
- Have a custom wood or stone type that is not included in vanilla Ex Nihilo: Sequentia (i.e., Oh The Biomes You'll Go or Biomes O'Plenty)
- Other conditions that may be added as discussion warrants and deems necessary (this conditions list will be modified as appropriate)

This thread will be used for requesting addons that meet at least one of the above criteria, or for discussion about why a mod should receive and addon rather than a data pack.

Any issues requesting compatibility will be close and directed to this thread.

**Comment by gruniu8** [comment]
will there be an addon that adds auto sieve and auto hammer like it was in 1.12.2?

**Reply by NovaMachina** [reply]
At the present moment there is no plan to create an addon for automation. However, there has been an increasing number of requests for one, so it will be considered.

**Reply by GamingInfinite** [reply]
(if one were keeping to the same format as previous ex nihilo versions, this would actually be part of an Ex Compressum extension, which is really the only case where this would be needed).

**Comment by GamingInfinite** [comment]
Ex Compressum like addon (compressed Crook, Hammer, etc.)
Also add KubeJs support, not just craft tweaker or, if there is already compatibility, please note how to use it in the documentation.

**Reply by NovaMachina** [reply]
KubeJs is already supported because Ex Nihilo uses the recipe system. You should be able to use the KubeJs wiki: https://mods.latvian.dev/books/kubejs/page/recipeeventjs
In particular, look at the section starting on line 80.

**Reply by GamingInfinite** [reply]
Sorry, yes I saw this and tried to figure out how to do it myself, however both times I tried I encountered an error.  Do you mind typing up a quick example of how I should be doing it?  Also again, please include it in the Ex Nihilo Documentation alongside the CraftTweaker Documentation, as I feel having an example in a centralized place will help modpack developers (or aspiring ones like myself) understand how we should be handling recipes for this particular mod in KubeJS

**Reply by MrEizy** [reply]
> Sorry, yes I saw this and tried to figure out how to do it myself, however both times I tried I encountered an error. Do you mind typing up a quick example of how I should be doing it? Also again, please include it in the Ex Nihilo Documentation alongside the CraftTweaker Documentation, as I feel having an example in a centralized place will help modpack developers (or aspiring ones like myself) understand how we should be handling recipes for this particular mod in KubeJS

That would really be nice like a example datapack recipe for all the different types of Functions in ex nihilo.
Same with removing recipes using datapacks.

**Reply by GamingInfinite** [reply]
> KubeJs is already supported because Ex Nihilo uses the recipe system. You should be able to use the KubeJs wiki: https://mods.latvian.dev/books/kubejs/page/recipeeventjs In particular, look at the section starting on line 80.

I finally figured out what I was doing wrong, but I still feel like you should note in the documentation how it should be laid out if you're not gonna create a custom function for your recipes (for instance event.sieve like the Create Mod has)

**Reply by NovaMachina** [reply]
Please make a suggestion in the issue collector of you would like to see this

**Comment by MrEizy** [comment]
It would be nice if you added a way in the config to disable all the DEFAULT drops from the sieves with a true or false option.

**Reply by GamingInfinite** [reply]
Actually, you CAN do this with KubeJS already.  You just write a single line of code to remove all recipes relating to sieves.  It would look something like this:

```
OnEvent('recipes', event => {
     event.remove({type: "exnihilosequentia:sieve"})
})
```

Literally nukes all recipes of that specific type.

**Comment by MrEizy** [comment]
Make a Ex-Nihilo Sequentia Tinkers addon where we can make hammers and crooks in tinkers like we could in 1.12

**Reply by NovaMachina** [reply]
This is a WIP. Should be a release soon for 1.18.

**Reply by NovaMachina** [reply]
This was released yesterday.

**Comment by GamingInfinite** [comment]
I personally think an in-house addon for Mystical Agriculture would be cool

**Reply by NovaMachina** [reply]
As far as I'm aware Mysical Agriculture does not have any ores that would require custom items from Ex Nihilo. I don't believe that it meets any of the requirements listed in the original post.

**Reply by GamingInfinite** [reply]
My main interaction with previous ex nihilo mods is through the sky factory series, in which the only way to obtain prosperity shards, the main crafting ingredient, which comes as ore, is through sieving sand.  My main thought was that their could possibly be built in integration with Mystical Agriculture to include inferium essence and prosperity shards as part of the drop table, however, by the way this is phrased I assume you expect that type of integration to be handled either on the side of mystical agriculture or of a pack maker of something similar to skyfactory.

**Reply by NovaMachina** [reply]
Correct. Since Minecraft added the ability for data packs this allows players, and pack creators, to add their own recipes. Ex Nihilo fully supports that system. Since I cannot know every compatibility that will be needed, I leave it to the players and pack creators to develop their own recipes when needed. The only time that an addon will be considered is when it is not possible for a data pack to add functionality, for example, the Thermal series adds three new mobs that would need a doll to be spawned via Ex Nihilo. Dolls cannot be created via a data pack and therefore an addon was created. Same with Mekanism and its Osmium ore.

**Comment by GamingInfinite** [comment]


**Reply by NovaMachina** [reply]
Please submit this to the issue collector. This thread is for add-ons only.

**Reply by NovaMachina** [reply]
Also, this functionality should already exist.

**Comment by sidjadon** [comment]
Does[ minecraft 1.18 ore distribution](https://gameinstants.com/minecraft-1-18-ore-distribution/) can help it

**Reply by remplerus** [reply]
what do you mean? i don't get what you want.
